azure/datalake/store/lib.py

```diff
@@ -68,7 +71,8 @@ def refresh_token(token):
 
 def auth(tenant_id=default_tenant, username=default_username,
          password=default_password, client_id=default_client,
-         client_secret=default_secret, resource=default_resource, **kwargs):
+         client_secret=default_secret, resource=default_resource,
+         require_2fa=False, **kwargs):
```

Comment:
I'm not a huge fan of this approach. There is no "default_password" and pushing into environment is not setting a default password for me.
I would prefer `password=None` in the signature and write inside the method:
```python
password = os.environ.get('azure_password', None) if not password
```